pkgs/build-support/go/module.nix

```diff
@@ -98,6 +98,8 @@ let
       runHook preConfigure
       export GOCACHE=$TMPDIR/go-cache
       export GOPATH="$TMPDIR/go"
+      # https://sourcehut.org/blog/2023-01-09-gomodulemirror/
+      export GOPRIVATE=git.sr.ht
```

Comment:
We need to make sure to not overwrite existing GOPRIVATE entries which could exist trough impureEnvs.

Reply:
Not sure I understand?
https://github.com/NixOS/nixpkgs/blob/bb7aafebb3086b5e0e7b80fad277c02cd4d86d13/pkgs/build-support/go/module.nix#L93-L95

Reply:
The only mention of GOPRIVATE I have been able to find is in a [home-manager module](https://github.com/nix-community/home-manager/blob/master/modules/programs/go.nix#L63-L73).

I don't think this would have impact on home-manager's stuff though (haven't extensively looked..).

Reply:
It won't, this is only set for the vendor FOD.

Reply:
I have a private packages that uses buildGoModule and sets the GOPRIVATE env to the internal github enterprise. This change would overwrite that.